src/pal/tests/palsuite/exception_handling/pal_sxs/test1/dlltest1.cpp

```diff
@@ -30,12 +39,14 @@ int DllTest1()
 
     PAL_TRY(VOID*, unused, NULL)
     {
+        EHGuard(NULL);
```

Comment:
This is fragile w.r.t. the possibility of future optimizations the compiler might attempt. It can decide that the region between the two EHGuard calls cannot throw and keep the EH info only for the two ranges where the EHGuard is called.
I have applied change like you did to my local testing app and I was able to break it e.g. by adding a variable of a type with destructor right before the second EHGuard.
The originally suggested way should be safe w.r.t. to these. The idea was to put the line that can throw (`*p = 1`) to a separate function marked with `__attribute__((noinline,optnone))` and adding the helper throw there, e.g. like this:
```C++
static __attribute__((noinline,optnone))
void FailingFunction(volatile void *p)
{
    if (p == NULL)
    {
        throw PAL_SEHException();
    }

    *p = 1;
}
```
Then replace the `*p = 1` in the original code by a call to this function.

Although this is just a test, people may look at it as an example of how to handle similar cases and so I would prefer this safer way.